### PR TITLE
[Debugger] Add evaluation support for XML filter and step expressions

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR 
- adds expression evaluation support for Ballerina filter and step expressions
- adds integration tests for related scenarios
- fixes https://github.com/ballerina-platform/ballerina-lang/issues/31640

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java [hunk 1]
```diff
@@ -363,12 +368,28 @@ public void visit(InterpolationNode interpolationNode) {
         interpolationNode.expression().accept(this);
     }
 
+    @Override
+    public void visit(XMLStepExpressionNode xMLStepExpressionNode) {
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java:372]:
```suggestion
    public void visit(XMLStepExpressionNode xmlStepExpressionNode) {
```

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java [hunk 2]
```diff
@@ -363,12 +368,28 @@ public void visit(InterpolationNode interpolationNode) {
...
+        visitSyntaxNode(xMLStepExpressionNode);
+        xMLStepExpressionNode.expression().accept(this);
+        Evaluator subExprEvaluator = result;
+        result = new XMLStepExpressionEvaluator(context, xMLStepExpressionNode, subExprEvaluator);
+    }
+
+    @Override
+    public void visit(XMLFilterExpressionNode xMLFilterExpressionNode) {
```
Comment (praveennadarajah) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java:380]:
```suggestion
    public void visit(XMLFilterExpressionNode xmlFilterExpressionNode) {
```

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java]: Fixed

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluatorBuilder.java]: Fixed

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**